### PR TITLE
feat: add vaultAddress and pm test

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
**Focus:**
Update vault addresses and add new chains to `VAULTS_CONFIG_BY_CHAIN`.

### Detailed summary
- Updated `vaultAddress` in various vault configurations.
- Added new chains to `VAULTS_CONFIG_BY_CHAIN`.
- Imported and used different lodash functions.
- Created a `publicClient` object based on chain IDs.
- Added tests for vault configurations and chain-specific tokens.

> The following files were skipped due to too many changes: `packages/position-managers/src/config.test.ts`, `packages/position-managers/src/constants/vaults/bsc.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/position-managers/src/config.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { chainNames } from '@pancakeswap/chains'
+import { Token } from '@pancakeswap/sdk'
+import difference from 'lodash/difference.js'
+import uniqBy from 'lodash/uniqBy.js'
+import { PublicClient, createPublicClient, http, parseAbiItem } from 'viem'
+import * as CHAINS from 'viem/chains'
+import { describe, expect, it } from 'vitest'
+import { SUPPORTED_CHAIN_IDS, VAULTS_CONFIG_BY_CHAIN } from './constants'
+
+const publicClient = Object.keys(VAULTS_CONFIG_BY_CHAIN).reduce((acc, chainId) => {
```
Comment (Chef-Yogi) [inline packages/position-managers/src/config.test.ts:10]: Thanks, man, 
do you think we should merge this test with 
"apps/web/src/config/__tests__/vaults.test.ts"

File: packages/position-managers/src/constants/vaults/bsc.ts [hunk 2]
```diff
@@ -91,11 +95,12 @@ export const vaults: VaultConfig[] = [
   },
   {
     id: 15,
-    idByManager: 2,
+    idByManager: 15,
```
Comment (chefjackson) [inline packages/position-managers/src/constants/vaults/bsc.ts:98]: @Chef-Yogi for the id by manager, I thought it's starting from 1 for each manager no?

Comment (Chef-Yogi) [reply packages/position-managers/src/constants/vaults/bsc.ts:98]: Yeah, ID by the manager starts from 1 for each manager, not.
Just for display

Comment (Chef-Yogi) [reply packages/position-managers/src/constants/vaults/bsc.ts:98]: so we don't expect that to be a unique key
CleanShot 2024-04-10 at 15 34 35@2x

**Review by Chef-Yogi — APPROVED** [review]

**Pull request merged**